src/main/java/com/squareup/catalog/demo/Main.java

```diff
@@ -233,29 +270,26 @@ private void executeExample(String exampleName, boolean cleanup, CatalogApi cata
   }
 
   /**
-   * Attempts to log {@link Error}s from an {@link ApiException}, or rethrows if the response body
-   * cannot be parsed.
+   * Finds out the currency to be used across all examples Use join() because we
+   * can't proceed further until we have this information.
+   *
+   * @param locationsApi the LocationsApi utility
    */
-  private void handleApiException(ApiException apiException) {
-    try {
-      // If the response includes a body, it means that he server returned some error message.
-      ErrorResponse response =
-          GsonProvider.gson().fromJson(apiException.getResponseBody(), ErrorResponse.class);
-
-      // If we found errors in the response body, log them. Otherwise, rethrow.
-      if (!checkAndLogErrors(response.errors, logger)) {
-        throw new RuntimeException(apiException);
+  private void setCurrencyAcrossApplication(LocationsApi locationsApi) {
+    locationsApi.listLocationsAsync().thenAccept(result -> {
+      if (checkAndLogErrors(result.getErrors(), logger)) {
+        return;
       }
-    } catch (JsonSyntaxException e) {
-      // If the error message isn't in JSON format, rethrow the error.
-      throw new RuntimeException(apiException);
-    }
-  }
 
-  /**
-   * Represents the response body of an {@link ApiException} that contains server specified errors.
-   */
-  private static class ErrorResponse {
-    List<Error> errors;
+      // grab the first location for the user, and use that to determine currency.
+      if (result.getLocations() != null && result.getLocations().size() > 0) {
```

Comment:
Should there be an `else` case here? Maybe default to USD, or throw an error?

Reply:
I believe this branch will never be taken. Each Merchant/Application is created with at least one location, which means this that result will always have at least 1 location. If the access token is incorrect or something goes wrong with the `listLocation` request, then the `exceptionally` portion of the code will be executed. To summarize, this if statement can probably be omitted altogether, I just didn't think it through. Up to you If you want it removed or I can add a dummy else branch with logging. In any case, I updated the logging in the case of failed `listLocation` call to emphasize that the issue might be with an incorrect access token.